docs/sources/reference/components/pyroscope/pyroscope.receive_http.md

```diff
@@ -0,0 +1,114 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/pyroscope/pyroscope.receive_http/
+aliases:
+  - ../pyroscope.receive_http/ # /docs/alloy/latest/reference/components/pyroscope.receive_http/
+description: Learn about pyroscope.receive_http
+title: pyroscope.receive_http
+---
+
+# pyroscope.receive_http
+
+`pyroscope.receive_http` listens for HTTP requests containing profiles and forwards them to other components capable of receiving profiles.
+
+The HTTP API exposed is compatible with Pyroscope's [HTTP ingest API](https://grafana.com/docs/pyroscope/latest/configure-server/about-server-api/). This allows `pyroscope.receive_http to act as a proxy for Pyroscope profiles, enabling flexible routing and distribution of profile data.
+
+## Usage
+
+```alloy
+pyroscope.receive_http "LABEL" {
+  http {
+    listen_address = "LISTEN_ADDRESS"
+    listen_port = PORT
+  }
+  forward_to = RECEIVER_LIST
+}
+```
+The component will start an HTTP server supporting the following endpoint:
+
+`POST /ingest` - send profiles to the component, which in turn will be forwarded to the receivers as configured in the `forward_to argument`. The request format must match that of Pyroscope's ingest API.
+
+## Arguments
+
+The following arguments are supported:
+
+Name              | Type          | Description                                     | Default | Required
+------------------|---------------|-------------------------------------------------|---------|---------
+`forward_to` | `list(ProfilesReceiver)` | List of receivers to send profiles to. |         | yes
+
+## Blocks
+
+The following blocks are supported inside the definition of `pyroscope.receive_http`:
+
+Hierarchy | Name | Description                                        | Required
+----------|------|----------------------------------------------------|---------
+`http`    | `http` | Configures the HTTP server that receives requests. | no
+
+### http
+
+The `http` block configures the HTTP server.
+
+You can use the following arguments to configure the `http` block. Any omitted fields take their default values.
+
+Name                   | Type       | Description                                                                                                      | Default  | Required
+-----------------------|------------|------------------------------------------------------------------------------------------------------------------|----------|---------
+`conn_limit`           | `int`      | Maximum number of simultaneous HTTP connections. Defaults to no limit.                                           | `0`      | no
+`listen_address`       | `string`   | Network address on which the server listens for new connections. Defaults to accepting all incoming connections. | `""`     | no
```

Comment:
Is this only HTTP or is HTTPS also supported? From the verbiage and underlying server I assume only HTTP?

Reply:
From Alloy's perspective, I believe this is acceptable since we have other receiver components (like Loki's [API component](https://github.com/grafana/alloy/blob/main/internal/component/loki/source/api/api.go#L32)) using HTTP-only. From our perspective, Pyroscope SDKs don't require TLS for pushing profiles, so we're fine on this side. I'd say we can add HTTPS support in the future if needed. 
Does this make sense to you?